test/tests/parallel/set2/test_helper.bats

```diff
@@ -97,29 +101,35 @@ load 'test_helper/common'
   assert_success
 }
 
-@test "wait_for_finished_setup_in_container" {
+@test "${TEST_NAME_PREFIX} wait_for_finished_setup_in_container" {
+  # the container name is used here explicitly, because it cannot be a local
+  # variable as the `teardown` function would not have the correct scope
+  # and does not see the variable
+
   # variable not local to make visible to teardown
   local PRIVATE_CONFIG
-  PRIVATE_CONFIG=$(duplicate_config_for_container .)
+  PRIVATE_CONFIG=$(duplicate_config_for_container . dms-test-wait_for_finished_setup)
 
-  CONTAINER_NAME=$(docker run -d --rm \
+  docker run -d --rm --name dms-test-wait_for_finished_setup \
     -v "${PRIVATE_CONFIG}":/tmp/docker-mailserver \
     -h mail.my-domain.com \
-    -t "${NAME}")
+    -t "${IMAGE_NAME}"
 
-  teardown() { docker rm -f "${CONTAINER_NAME}"; }
+  teardown() { docker rm -f dms-test-wait_for_finished_setup; }
 
   # the setup should not be finished immediately after starting
-  ! TEST_TIMEOUT_IN_SECONDS=0 wait_for_finished_setup_in_container "${CONTAINER_NAME}"
+  ! TEST_TIMEOUT_IN_SECONDS=0 wait_for_finished_setup_in_container dms-test-wait_for_finished_setup
+
+  docker exec dms-test-wait_for_finished_setup setup email add 'example_user@domain.com' 'password'
```

Comment:
Is this needed due to the changed `PRIVATE_CONFIG`?

Reply:
Yes, it is :)

Reply:
Reverted the explicit naming, and therefore this change as well :)